generative/maisi/scripts/diff_model_infer.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import os
+import random
+import torch
+import torch.distributed as dist
+import nibabel as nib
+import numpy as np
+
+from datetime import datetime, timedelta
+from tqdm import tqdm
+
+from monai.inferers import sliding_window_inference
+from monai.utils import set_determinism
+
+from utils import define_instance, load_autoencoder_ckpt
+
+
+class ReconModel(torch.nn.Module):
+    def __init__(self, autoencoder, scale_factor):
+        super().__init__()
+        self.autoencoder = autoencoder
+        self.scale_factor = scale_factor
+
+    def forward(self, z):
+        recon_pt_nda = self.autoencoder.decode_stage_2_outputs(z / self.scale_factor)
+        return recon_pt_nda
+
+
+def diff_model_infer(env_config_path: str, model_config_path: str):
```

Comment:
Overall I would suggest to break this function into smaller sub-function, which may improve the readability. Thanks!

Reply:
Hi @mingxin-zheng, I have updated the script. Please proceed. Thanks.